src/main/java/com/cloudlabs/server/Modules/ModuleController.java

```diff
@@ -0,0 +1,166 @@
+package com.cloudlabs.server.modules;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.cloudlabs.server.modules.Module;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import javax.swing.JSpinner.ListEditor;
+
+@CrossOrigin(origins = {"${app.security.cors.origin}"})
+@RestController
+@RequestMapping("/Modules")
+public class ModuleController {
+
+    @Autowired
+    private ModuleRepository repository;
+
+    /* 
+    @EventListener(ApplicationReadyEvent.class)
+    public void runAfterStartup() {
+        
+        Module newModule = new Module();
+        newModule.setModuleSubtitle("1_EH_000");
+        newModule.setModuleName("Ethical Hacking");
+        newModule.setModuleDescription("Explore various methods to extract valuable information from targeted systems without causing harm and gain practical experience in executing controlled attacks to simulate real-world scenarios, enabling the identification and mitigation of security weaknesses.");
+        this.repository.save(newModule);
+    }
+    */
+
+    @GetMapping
+    public List<Module> getAllModules(){
+        return this.repository.findAll();
+    }
+
+    @GetMapping("/{moduleId}")
+    public Module getModuleById(@PathVariable String moduleId) {
+        // Convert moduleId to Long
+        Long moduleIdAsLong;
+        try {
+            moduleIdAsLong = Long.valueOf(moduleId);
+        } catch (NumberFormatException e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid module ID format");
+        }
+
+        // Find Module by its id
+        Module module = this.repository.findByModuleId(moduleIdAsLong);
```

Comment:
Business logic and entities should not be inside the controller. Do seperate the logic into a different file, i.e. ModuleService and ModuleServiceImpl, you may refer to #19.